### PR TITLE
Add .name_repair to expand functions

### LINKED ISSUE #798: crossing() does not respect .name_repair
It seems that `crossing()` either does not respect the value of `.name_repair` or does not see this parameter:

``` r
vals <- c("v", "h")
tidyr::crossing(vals, vals, vals, vals, .name_repair = "unique")
#> Error: Column names `vals`, `vals`, `vals` must not be duplicated.
#> Use .name_repair to specify repair.
```

<sup>Created on 2019-11-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

`tibble()` does not fail in this situation:

``` r
vals <- c("v", "h")
tidyr::tibble(vals, vals, vals, vals, .name_repair = "unique")
#> New names:
#> * vals -> vals...1
#> * vals -> vals...2
#> * vals -> vals...3
#> * vals -> vals...4
#> # A tibble: 2 x 4
#>   vals...1 vals...2 vals...3 vals...4
#>   <chr>    <chr>    <chr>    <chr>   
#> 1 v        v        v        v       
#> 2 h        h        h        h
```

<sup>Created on 2019-11-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### PR DESCRIPTION
**Pull request by jeffreypullin**
Fixes #798 (Marked as help wanted) 

Notes: 
* Added .name_repair to `nesting` as well as the functions you mentioned in your comment on #798. 
* I wasn't sure how best to include the info about .name_repair in the roxygen so I ended up just copy and pasting it in. 

Very happy to receive any feedback!

Best, 
Jeffrey

### CODE REVIEW

File: R/expand.R [hunk 1]
```diff
@@ -28,6 +28,19 @@
 #'   `year = 2010:2020` or `year = \link{full_seq}(year,1)`.
 #'
 #'   Length-zero (empty) elements are automatically dropped.
+#' @param .name_repair Used to check that output data frame has valid
```
Comment (hadley) [inline R/expand.R]: Can you please use `@inheritParams` instead? (https://roxygen2.r-lib.org/articles/rd.html#inheriting-documentation-from-other-topics)

File: tests/testthat/test-expand.R [hunk 2]
```diff
@@ -126,3 +126,44 @@ test_that("crossing handles list columns", {
...
+  expect_named(out, c("x...1", "x...2"))
+
+  out <- expand_grid(x, x, .name_repair = "minimal")
+  expect_named(out, c("x", "x"))
+})
+
+test_that("crossing/nesting/expand respect .name_repair", {
+
+  x <- 1:2
+  expect_named(
```
Comment (hadley) [inline tests/testthat/test-expand.R]: Could you please fit each of these on a single line?

File: R/expand.R (hunk 1 above)
Comment (jeffreypullin) [reply R/expand.R]: `expand()` now inherits `.name_repair` from `expand_grid()`.

File: tests/testthat/test-expand.R (hunk 2 above)
Comment (jeffreypullin) [reply tests/testthat/test-expand.R]: Done

**Comment by jeffreypullin** [comment]
I also took the liberty of adding a NEWS bullet.

**Comment by hadley** [comment]
Thanks!!

**Pull request merged**